### PR TITLE
Fix up doc block return types for chaining.

### PR DESCRIPTION
**Pull request by dereuromark**
Implements https://github.com/cakephp/cakephp/pull/10018 respecting the static factories which actually have been correctly made non-$this returning.
@antograssiot @thinkingmedia  Please double check.

### CODE REVIEW

**Comment by dereuromark** [comment]
What does this actually return in the EventManager, self or static?
```
     * @return \Cake\Event\EventManager the global event manager
     */
    public static function instance($manager = null)
```

File: src/Console/ConsoleOptionParser.php [hunk 1]
```diff
@@ -173,7 +173,7 @@ public function __construct($command = null, $defaultOptions = true)
      */
     public static function create($command, $defaultOptions = true)
     {
-        return new ConsoleOptionParser($command, $defaultOptions);
+        return new static($command, $defaultOptions);
```
Comment (dereuromark) [inline src/Console/ConsoleOptionParser.php]: Is it then `@return self` or `@return static`?

**Comment by ADmad** [comment]
> What does this actually return in the EventManager, self or static?

If `EventManager::intance()` was first used as a setter to set an instance of a child class then it would return that instance when used as getter, in which case `return self` would be wrong. So `return static` makes more sense.

But in that case the line `static::$_generalManager = new EventManager();` should also probably be changed to ``static::$_generalManager = new static();``

File: src/Database/Query.php [hunk 2]
```diff
@@ -449,7 +449,7 @@ public function modifier($modifiers, $overwrite = false)
      *  passed as an array of strings, array of expression objects, or a single string. See
      *  the examples above for the valid call types.
      * @param bool $overwrite whether to reset tables with passed list or not
-     * @return self
+     * @return $this
```
Comment (chinpei215) [inline src/Database/Query.php]: Not your fault, but this should be `$this|array`.

Comment (dereuromark) [inline src/Database/Query.php]: You are right - fixed.

File: src/Database/Query.php [hunk 3]
```diff
@@ -556,7 +554,7 @@ public function from($tables = [], $overwrite = false)
      * @param array $types associative array of type names used to bind values to query
      * @param bool $overwrite whether to reset joins with passed list or not
      * @see \Cake\Database\Type
-     * @return self
+     * @return $this
```
Comment (chinpei215) [inline src/Database/Query.php]: The return type of `join()` also should be `$this|array`.

Comment (dereuromark) [inline src/Database/Query.php]: Oh, yeah, indeed. We definitely need to fix that up soon, as well. Thanks, I adjusted the doc block.

**Comment by dereuromark** [comment]
So, this should be good to go then?

File: src/Console/ConsoleOptionParser.php (hunk 1 above)
Comment (markstory) [inline src/Console/ConsoleOptionParser.php]: I think this is fine. It should break backwards compatibility unless there is an incompatible subclass which is not our purview.

**Review by markstory — COMMENTED** [review]
Looks good to me.

**Pull request merged**